scipy/signal/tests/test_filter_design.py

```diff
@@ -127,18 +128,18 @@ def test_output_order(self):
              3+1j, 3+1j, 3+1j, 3-1j, 3-1j, 3-1j,
              2-3j, 2+3j]
         zc, zr = _cplxreal(a)
-        assert_allclose(zc, [1j, 1j, 1j, 1+1j, 1+2j, 2+3j, 2+3j, 3+1j, 3+1j,
+        xp_assert_close(zc, [1j, 1j, 1j, 1+1j, 1+2j, 2+3j, 2+3j, 3+1j, 3+1j,
                              3+1j])
-        assert_allclose(zr, [0, 0, 1, 2, 3, 4])
+        xp_assert_close(zr, [0.0, 0, 1, 2, 3, 4])
 
         z = array([1-eps + 1j, 1+2j, 1-2j, 1+eps - 1j, 1+eps+3j, 1-2*eps-3j,
                    0+1j, 0-1j, 2+4j, 2-4j, 2+3j, 2-3j, 3+7j, 3-7j, 4-eps+1j,
                    4+eps-2j, 4-1j, 4-eps+2j])
 
         zc, zr = _cplxreal(z)
-        assert_allclose(zc, [1j, 1+1j, 1+2j, 1+3j, 2+3j, 2+4j, 3+7j, 4+1j,
+        xp_assert_close(zc, [1j, 1+1j, 1+2j, 1+3j, 2+3j, 2+4j, 3+7j, 4+1j,
                              4+2j])
-        assert_equal(zr, [])
+        assert_equal(zr, np.asarray([], dtype=zr.dtype))
```

Comment:
I suppose this is okay, but comparing `zr`'s dtype against `zr.dtype` isn't really doing much. Ideally the desired dtype is derived from the inputs.

Reply:
OK, changed in the last commit. Note however that here it works without explicit dtype only because `zr.dtype == np.float64`. Were it, say, integer, the assertion barfs on "mismatched dtype"